client/components/loading/StepContainerV2Loading.tsx

```diff
@@ -0,0 +1,24 @@
+import { Step } from '@automattic/onboarding';
+import { type ReactNode, useEffect, useState } from 'react';
+
+export const StepContainerV2Loading = ( {
+	title,
+	progress,
+	delay = 0,
+}: {
+	title?: ReactNode;
+	progress?: number;
+	delay?: number;
+} ) => {
+	const [ shouldDisplayTitle, setShouldDisplayTitle ] = useState( delay === 0 );
+
+	useEffect( () => {
+		const timeout = setTimeout( () => {
+			setShouldDisplayTitle( true );
+		}, delay );
+
+		return () => clearTimeout( timeout );
```

Comment:
Since `delay === 0` is supposed to bypass the the delaying effect 🤷

```suggestion
		if ( delay !== 0 ) {
			const timeout = setTimeout( () => {
				setShouldDisplayTitle( true );
			}, delay );

			return () => clearTimeout( timeout );
		}
```